components/Email/Email.js

```diff
@@ -19,8 +19,8 @@ import {
 } from '@chakra-ui/core';
 
 export default function Email() {
-	const { activeList, message, handleActionOnMail } = useContext(UserContext);
-	const { contract } = useContext(Web3Context);
+	const { activeList, message, handleActionOnMail, handleActionOnMailGasless, isGasless } = useContext(UserContext);
```

Comment:
Please keep components unaware of gas/gasless modes. Handle it in userState. Components should just trigger the operation.